data/sovereignTx/sovereignTx.go

```diff
@@ -0,0 +1,57 @@
+//go:generate protoc -I=. -I=$GOPATH/src -I=$GOPATH/src/github.com/multiversx/protobuf/protobuf  --gogoslick_out=. sovereignTx.proto
+package sovereignTx
+
+import (
+	"github.com/multiversx/mx-chain-core-go/data"
+	"math/big"
+)
+
+var _ = data.TransactionHandler(&SovereignTx{})
+
+// IsInterfaceNil verifies if underlying object is nil
+func (stx *SovereignTx) IsInterfaceNil() bool {
+	return stx == nil
+}
+
+// SetValue sets the value of the sovereign transaction
+func (stx *SovereignTx) SetValue(value *big.Int) {
+	stx.Value = value
+}
+
+// SetData sets the data of the sovereign transaction
+func (stx *SovereignTx) SetData(data []byte) {
+	stx.Data = data
+}
+
+// SetRcvAddr sets the receiver address of the sovereign transaction
+func (stx *SovereignTx) SetRcvAddr(addr []byte) {
+	stx.RcvAddr = addr
+}
+
+// SetSndAddr sets the sender address of the sovereign transaction
+func (stx *SovereignTx) SetSndAddr(addr []byte) {
+	stx.SndAddr = addr
+}
+
+// GetRcvUserName returns nil for sovereign transaction
+func (stx *SovereignTx) GetRcvUserName() []byte {
```

Comment:
[question]
Perhaps we could actually add the rcv username in the proto?
Thinking about it from the perspective of a "multiverse" where a user can "travel" worlds/sovereign shards

Reply:
I just kept it simple as the model of SCR, where these fields are not in (SndUserName and RcvUsernName). I think we can add them later if we decide that these fields are usable.